kudu/src/main/java/com/yahoo/ycsb/db/KuduYCSBClient.java

```diff
@@ -127,21 +136,82 @@ private void initClients() throws DBException {
 
       Properties prop = getProperties();
 
-      String masterAddresses = prop.getProperty(MASTER_ADDRESSES_OPT,
+      final String masterAddresses = prop.getProperty(MASTER_ADDRESSES_OPT,
                                                 "localhost:7051");
       LOG.debug("Connecting to the masters at {}", masterAddresses);
 
-      int numClients = getIntFromProp(prop, NUM_CLIENTS_OPT, DEFAULT_NUM_CLIENTS);
-      for (int i = 0; i < numClients; i++) {
-        clients.add(new KuduClient.KuduClientBuilder(masterAddresses)
-                                  .defaultSocketReadTimeoutMs(DEFAULT_SLEEP)
-                                  .defaultOperationTimeoutMs(DEFAULT_SLEEP)
-                                  .defaultAdminOperationTimeoutMs(DEFAULT_SLEEP)
-                                  .build());
+      try {
+        int numClients = getIntFromProp(prop, NUM_CLIENTS_OPT, DEFAULT_NUM_CLIENTS);
+        UserGroupInformation loginUser = getKerberosUser(prop);
+        for (int i = 0; i < numClients; i++) {
+          if (loginUser != null) {
+            // Kerberos connection
+            clients.add(loginUser.doAs(
+                new PrivilegedExceptionAction<KuduClient>() {
+                  @Override
+                  public KuduClient run() {
+                    return new KuduClient.KuduClientBuilder(masterAddresses)
+                        .defaultSocketReadTimeoutMs(DEFAULT_SLEEP)
+                        .defaultOperationTimeoutMs(DEFAULT_SLEEP)
+                        .defaultAdminOperationTimeoutMs(DEFAULT_SLEEP)
+                        .build();
+                  }
+                }));
```

Comment:
Instead of duplicating the builder incantation twice, what if you assign the PrivilegedExceptionAction to a variable, and then just call 'action.run()' vs 'loginUser.doAs(action.run())'?

Also, I think if you just use UserGroupInformation.getLoggedInUser() or somesuch you are guaranteed to get _some_ user which can do a doAs, so you might not need this conditional at all